internal/provider/kubernetes/predicates.go

```diff
@@ -288,3 +288,25 @@ func (r gatewayAPIReconciler) findOwningGateway(ctx context.Context, labels map[
 
 	return gtw
 }
+
+func (r *gatewayAPIReconciler) handleNode(obj client.Object) bool {
+	ctx := context.Background()
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		r.log.Info("unexpected object type, bypassing reconciliation", "object", obj)
+		return false
+	}
+
+	key := types.NamespacedName{Name: node.Name}
+	if err := r.client.Get(ctx, key, node); err != nil {
+		if kerrors.IsNotFound(err) {
+			utils.ProviderStore().RemoveNode(node)
```

Comment:
so the way to know the obj was deleted, is to perform a `GET` and see if the obj exists or not ?

Reply:
yes, we don't get the event type in reconciler.